internal/frontend/middleware/global.go

```diff
@@ -95,15 +98,21 @@ func Setup(opts *Options) {
 	authBasic = opts.AuthBasic
 	authToken = opts.AuthToken
 	appLogger = opts.Logger
+	basePath = opts.BasePath
 }
 
 func prefixChecker(next http.Handler) http.Handler {
 	return http.HandlerFunc(
 		func(w http.ResponseWriter, r *http.Request) {
-			if strings.HasPrefix(r.URL.Path, "/api") {
-				next.ServeHTTP(w, r)
+			if basePath != "" && r.URL.Path == "/" {
+				http.Redirect(w, r, basePath, http.StatusSeeOther)
+				return
+			}
+
+			if strings.HasPrefix(r.URL.Path, path.Join(basePath, "/api")) {
+				http.StripPrefix(basePath, next).ServeHTTP(w, r)
 			} else {
-				defaultHandler.ServeHTTP(w, r)
+				http.StripPrefix(basePath, defaultHandler).ServeHTTP(w, r)
```

Comment:
`http.StripPrefix` seemed like the cleanest way to allow all existing routes to work as before. 

This is my first time writing Go so if there's a more idiomatic way of doing this let me know!